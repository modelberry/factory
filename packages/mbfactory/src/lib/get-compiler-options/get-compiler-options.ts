import ts from 'typescript'

export const getCompilerOptions = () => {
  const configFileName = ts.findConfigFile(
    './',
    ts.sys.fileExists,
    // TODO: Make this a config option
    'tsconfig.json'
  )
  const configFile = ts.readConfigFile(configFileName!, ts.sys.readFile)
  const compilerOptions = ts.parseJsonConfigFileContent(
    configFile.config,
    ts.sys,
    './'
  )
  return compilerOptions
}
