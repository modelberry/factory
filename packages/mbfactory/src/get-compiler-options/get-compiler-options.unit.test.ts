import { getCompilerOptions } from './get-compiler-options'

describe('getCompilerOptions', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  test('should get compiler options', async () => {
    expect(getCompilerOptions().options).toMatchSnapshot()
  })
})
