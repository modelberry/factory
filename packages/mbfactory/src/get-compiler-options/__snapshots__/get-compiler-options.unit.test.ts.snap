// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCompilerOptions should get compiler options 1`] = `
{
  "allowJs": true,
  "allowUmdGlobalAccess": true,
  "configFilePath": undefined,
  "declaration": false,
  "downlevelIteration": true,
  "esModuleInterop": true,
  "forceConsistentCasingInFileNames": true,
  "jsx": 4,
  "lib": [
    "lib.dom.d.ts",
    "lib.dom.iterable.d.ts",
    "lib.esnext.d.ts",
  ],
  "module": 1,
  "moduleResolution": 3,
  "noImplicitReturns": true,
  "noUnusedLocals": true,
  "removeComments": true,
  "resolveJsonModule": true,
  "skipLibCheck": true,
  "sourceMap": false,
  "strict": true,
  "target": 1,
}
`;
